roles/client/tasks/main.yml

```diff
@@ -22,15 +22,43 @@
   retries: 5
   when: ursula_os == 'rhel' and openstack_install_method == 'distro'
 
+- block:
+  - name: make sure /opt/stack/ exists
+    file: dest=/opt/stack state=directory
+
+  - name: write constraints file
+    copy:
+      src: ../../openstack-source/files/constraints.txt
+      dest: /opt/stack/constraints.txt
+      force: true
+    when: client.upper_constraints == None
```

Comment:
Should have conditionals check for "is defined" or default them to None. My comment applies to all the condtionals checking for upper_constraints vars